apps/backend/src/modules/users/user.repository.ts

```diff
@@ -60,7 +69,17 @@ class UserRepository implements Repository {
 	}
 
 	public async findByEmail(email: string): Promise<null | UserEntity> {
-		const user = await this.userModel.query().findOne({ email });
+		const user = await this.userModel
+			.query()
+			.findOne({ email })
+			.whereNull("deletedAt")
+			.execute();
+
+		return user ? UserEntity.initialize(user) : null;
+	}
+
+	public async findByEmailCreate(email: string): Promise<null | UserEntity> {
+		const user = await this.userModel.query().findOne({ email }).execute();
```

Comment:
Don't we need to add a check with deleted at here?

Reply:
We don`t need a check when we use this method in sign up, so user can`t sign up with the soft deleted email. I will change this methods, I think I made it wrong.